File: arkts_static_test/02_LexicalElements/ets_cases/2.9.8_Undefined_Literal/cross_lang_verify/UndefinedLiteralRuntimeTest.java
/*
* Copyright (c) 2021-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
// Java Runtime Test - 对应 2.9.8 Undefined Literal
// 测试重点：undefined 值、比较、类型检查
// 覆盖 3 个 runtime 测试场景 (006-008)

public class UndefinedLiteralRuntimeTest {
    public static void main(String[] args) {
        int totalAssertions = 0;

        System.out.println("[Java] Undefined Literal Runtime Test");
        System.out.println("[Java] NOTE: Java uses 'null' instead of 'undefined'");
        System.out.println("");

        // 006: undefined 值验证
        // Java 没有 undefined，使用 null 模拟
        Object x1 = null;
        assert x1 == null : "006: null failed";
        totalAssertions++;
        System.out.println("[Java] 006 null value: PASSED (" + x1 + ")");

        // 007: undefined 比较验证
        Object x2 = null;
        Object x3 = null;
        assert x2 == x3 : "007: null equality failed";
        totalAssertions++;
        System.out.println("[Java] 007 null equality: PASSED");

        // null 与数字比较
        int x4 = 0;
        assert x2 != (Object) x4 : "007: null should not equal 0";
        totalAssertions++;
        System.out.println("[Java] 007 null != 0: PASSED");

        // null 与字符串比较
        String x5 = "null";
        assert !x5.equals(x2) : "007: null should not equal \"null\"";
        totalAssertions++;
        System.out.println("[Java] 007 null != \"null\": PASSED");

        // 008: undefined 类型检查
        Object x6 = null;
        assert x6 == null : "008: null check failed";
        totalAssertions++;
        System.out.println("[Java] 008 null check: PASSED");

        System.out.println("=== Java Undefined Literal Runtime Test PASSED ===");
        System.out.println("Total assertions passed: " + totalAssertions);
    }
}

File: arkts_static_test/02_LexicalElements/ets_cases/2.9.8_Undefined_Literal/cross_lang_verify/UndefinedLiteralRuntimeTest.swift
/*
* Copyright (c) 2021-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
// Swift Runtime Test - 对应 2.9.8 Undefined Literal
// 测试重点：undefined 值、比较、类型检查
// 覆盖 3 个 runtime 测试场景 (006-008)

import Foundation

@main
struct UndefinedLiteralRuntimeTest {
    static func main() {
        print("=== Swift Undefined Literal Runtime Test ===")
        print("[Swift] NOTE: Swift uses 'nil' instead of 'undefined'")
        print("")

        var totalAssertions = 0

        // 006: undefined 值验证
        // Swift 没有 undefined，使用 nil 模拟
        var x1: Int? = nil
        assert(x1 == nil, "006: nil failed")
        totalAssertions += 1
        print("[Swift] 006 nil value: PASSED (\(String(describing: x1)))")

        // 007: undefined 比较验证
        var x2: Int? = nil
        var x3: Int? = nil
        assert(x2 == x3, "007: nil equality failed")
        totalAssertions += 1
        print("[Swift] 007 nil equality: PASSED")

        // nil 与数字比较
        let x4 = 0
        assert(x2 != x4, "007: nil should not equal 0")
        totalAssertions += 1
        print("[Swift] 007 nil != 0: PASSED")

        // nil 与字符串比较
        let x5: String? = nil
        assert(x5 == nil, "007: string nil == nil failed")
        totalAssertions += 1
        print("[Swift] 007 string nil == nil: PASSED")

        // 008: undefined 类型检查
        var x6: Int? = nil
        assert(x6 == nil, "008: nil check failed")
        totalAssertions += 1
        print("[Swift] 008 nil check: PASSED")

        print("")
        print("=== Swift Undefined Literal Runtime Test PASSED ===")
        print("Total assertions passed: \(totalAssertions)")
    }
}
